src/telemetry/http_service.rs

```diff
@@ -0,0 +1,91 @@
+use std::{
+	pin::Pin,
+	sync::{Arc, Mutex},
+	task::{Context, Poll},
+};
+
+use futures::Future;
+use hyper::{service::Service, Body, Method, Request, Response, StatusCode};
+use prometheus_client::{encoding::text::encode, registry::Registry};
+
+type SharedRegistry = Arc<Mutex<Registry>>;
+
+pub struct MetricService {
+	reg: SharedRegistry,
+}
+
+impl MetricService {
+	fn get_reg(&mut self) -> SharedRegistry {
+		Arc::clone(&self.reg)
+	}
+
+	fn respond_with_metrics(&mut self) -> Response<Body> {
+		let mut encoded: Vec<u8> = Vec::new();
+		let reg = self.get_reg();
+
+		encode(&mut encoded, &reg.lock().unwrap()).unwrap();
+		let metrics_content_type = "application/openmetrics-text;charset=utf-8;version=1.0.0";
```

Comment:
We could try to serve just a simple json for ease of use on the `/metric` endpoint, and create a separate endpoint for OM. This content type triggers file download in browsers, which is not a desired outcome for further prometheus integration (ie. Datadog).